processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -1173,14 +1173,20 @@ public ExpressionType getOutputType(Expr.InputBindingInspector inspector, List<E
       );
     }
 
+    @Override
+    public boolean canVectorize(Expr.InputBindingInspector inspector, List<Expr> args)
+    {
+      return false;
+    }
+
     @Override
     protected ExprEval eval(final long x, final long y)
     {
       if (y == 0) {
         if (x != 0) {
           return ExprEval.ofLong(null);
         }
-        return ExprEval.ofLong(0);
+        return ExprEval.ofLong(NullHandling.defaultLongValue());
```

Comment:
why do we need to distinguish case of numerator being non-zero? in sql compatible mode this results in null in both cases. 

Also, i think this should just be set to `ExprEval.ofLong(null)` in both cases since after #13809 the coercion of null to default values happens when stuff gets out of the expression layer instead of inside it